phpstan.neon.dist

```diff
@@ -13,10 +13,15 @@ parameters:
     treatPhpDocTypesAsCertain: false
     checkMissingVarTagTypehint: true
     checkMissingTypehints: true
+    # @todo: Remove the "checkGenericClassInNonGenericObjectType" definition.
+    checkGenericClassInNonGenericObjectType: false
     ignoreErrors:
         - '#^Property Gedmo\\Tests\\.+\\Fixture\\[^:]+::\$\w+ is never written, only read\.$#'
         - '#^Property Gedmo\\Tests\\.+\\Fixture\\[^:]+::\$\w+ is never read, only written\.$#'
         - '#^Property Gedmo\\Tests\\.+\\Fixture\\[^:]+::\$\w+ is unused\.$#'
+        # @todo: Remove the following ignored error when the parameters are correctly declared.
+        - '#^Method Gedmo\\[^:]+::[^\(]+\(\) has parameter \$\w+ with no value type specified in iterable type array\.$#'
```

Comment:
Which errors are ignored by this one?

Reply:
The lack of definition on the iterable arguments in the methods. By instance:
```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/Gedmo/Tool/BaseTestCaseOM.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  74     Method Gedmo\Tests\Tool\BaseTestCaseOM::getMongoDBDriver() has parameter $paths with no value type specified in iterable type array.                      
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
  74     Method Gedmo\Tests\Tool\BaseTestCaseOM::getMongoDBDriver() has parameter $paths with no value type specified in iterable type array.                      
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
  83     Method Gedmo\Tests\Tool\BaseTestCaseOM::getORMDriver() has parameter $paths with no value type specified in iterable type array.                          
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
  83     Method Gedmo\Tests\Tool\BaseTestCaseOM::getORMDriver() has parameter $paths with no value type specified in iterable type array.                          
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
  113    Method Gedmo\Tests\Tool\BaseTestCaseOM::getDefaultMockSqliteEntityManager() has parameter $fixtures with no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
  113    Method Gedmo\Tests\Tool\BaseTestCaseOM::getDefaultMockSqliteEntityManager() has parameter $fixtures with no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
```

Reply:
is it only under the `tests` directory? or everywhere? I was thinking if the regex should be more specific, otherwise seems like it will ignore any iterable argument

Reply:
It is everywhere. In fact, there are 278 (257 under `src/`) errors ignored by this rule.
Based on your concern, I think we could use the baseline instead; although my intention is not to retain this rule for long time.

Reply:
no worries then, the baseline IIRC is manually modified so it will be difficult to update it